src/MultiPartStreamFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Message;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Factory for creating Multipart Streams.
+ *
+ * @author Robert Schönthal <robert.schoenthal@gmail.com>
+ */
+interface MultiPartStreamFactory
+{
+    /**
+     * Creates a Multipart Stream from parameters and/or files.
+     *
+     * @param array $parameters
+     * @param array $files
+     *
+     * @return StreamInterface
+     *
+     * @throws \InvalidArgumentException If the $parameters or $files are invalid.
```

Comment:
What does invalid mean here? I think this suggests that we are not completely sure about what parameters/files do we accept. We should clear that concept and make it clear all over the documentation (both phpdoc and rtd)